spring-boot-project/spring-boot-autoconfigure/build.gradle

```diff
@@ -58,6 +58,11 @@ dependencies {
 	}
 	optional("org.apache.httpcomponents.client5:httpclient5")
 	optional("org.apache.kafka:kafka-streams")
+	optional("org.apache.pulsar:pulsar-client-reactive-producer-cache-caffeine") {
```

Comment:
This transitively brings in caffeine. There is WIP to shade the fact that the cache mechanism is using caffeine. It is an internal detail and not intended to swapped out.

Reply:
https://github.com/apache/pulsar-client-reactive/pull/124